NBitcoin.Tests/RPCClientTests.cs

```diff
@@ -1404,8 +1406,8 @@ public void ShouldWalletProcessPSBTAndExtractMempoolAcceptableTX()
 				result = client.WalletProcessPSBT(psbtUnFinalized, false, type, bip32derivs: true);
 				Assert.False(result.Complete);
 				Assert.False(result.PSBT.CanExtractTX());
-				result.PSBT.TryFinalize(out bool isFinalized2);
-				Assert.False(isFinalized2);
+				result.PSBT.TryFinalize(out var errors2);

```

Comment:
I have changed the `TryFinalize` API to take `out InvalidOperationException[]` instead of `out bool` so that the user can see why (and which input) is failing to finalize (breaking change). So `Assert.NotEmpty` below is equal to previous `Assert.False`
Sorry that I have put too many changes in one PR, I will split.